### PR TITLE
Fix Lora meta data propagation

### LINKED ISSUE #3331: Native executable based Lora adapter encodes JSON incorrectly
The `meta_data` encoded by the lorawan adapter is not encoded properly when running in native mode, it is encoded as  `{}` and thus without conents.

### PR DESCRIPTION
**Pull request by sophokles73**
The value objects used by the Lora adapter have been annotated to
register them for reflection when compiling a native executable.

Fixes #3331

### CODE REVIEW

**Comment by BobClaerhout** [comment]
> @BobClaerhout maybe you want to take a look as well ...

Thanks for the ping. It's fairly easy it seems but you need to know what's required... I especially like the addition of the automated tests :+1:

**Comment by sophokles73** [comment]
> I especially like the addition of the automated tests +1

Yes, if only now we had some proper documentation for the adapter ...

File: tests/src/test/java/org/eclipse/hono/tests/lora/LoraIT.java [hunk 1]
```diff
@@ -0,0 +1,773 @@
...
+     * Creates an HTTP Basic Authorization header value for a device.
+     * <p>
+     * The credentials are Base64 encoded in the Basic auth header's username segment.
+     *
+     * @param tenant The tenant that the device belongs to.
+     * @param deviceId The device identifier.
+     * @param password The device's password.
+     * @return The header value.
+     */
+    protected static String getBasicAuthWithEncodedCredentials(final String tenant, final String deviceId, final String password) {
```
Comment (calohmn) [inline tests/src/test/java/org/eclipse/hono/tests/lora/LoraIT.java:765]: Method not being used.

**Review by calohmn — APPROVED** [review]
LGTM

**Pull request merged**